CMakeExt/MPI.cmake

```diff
@@ -35,3 +37,18 @@ if (MPI_INCLUDE_PATH AND MPI_LIBRARY)
 else (MPI_INCLUDE_PATH AND MPI_LIBRARY)
   set(MPI_FOUND FALSE CACHE BOOL "Did not find the MPI library")
 endif (MPI_INCLUDE_PATH AND MPI_LIBRARY)
+
+
+# check for MPI-3
+
+set(CMAKE_REQUIRED_INCLUDES ${MPI_INCLUDE_PATH})
```

Comment:
Just found `CMAKE_PUSH_CHECK_STATE` and `CMAKE_POP_CHECK_STATE` in `CMakePushCheckState.cmake`. I propose to surround the set and check with these two calls.